source/dofs/dof_renumbering.cc

```diff
@@ -689,7 +690,12 @@ namespace DoFRenumbering
                                             dof_handler.end(),
                                             component_order_arg,
                                             false);
-    if (result == 0)
+    (void)result;
+
+    // If we don't have a renumbering (i.e., when there is 1 component) then
+    // return
+    if (Utilities::MPI::max(renumbering.size(),
+                            dof_handler.get_communicator()) == 0)
       return;
```

Comment:
I don't understand what you are doing here. The `renumbering` array has a nonzero size, how can its maximum be zero?

Reply:
(That implies that I don't understand what the function was doing before already.)

Reply:
`DoFRenumbering::component_wise()` can set the array of new indices to have a length of zero in two circumstances:
1. There are no locally owned DoFs
2. There is only one component and no renumbering needs to be done

so I had to add those calls to `max()` everywhere to make sure we cover the second case.

Reply:
Ah, I see. Thanks for the explanation!